File: clang/docdb_test.cpp
#include "docdb.hpp"
#include <iostream>

using namespace std;

int main() {
	int packageId = -1, sourceId = -1, namespaceId;
	DocDb docDb("localhost", "ir", "ir", "ir");

	if (docDb.connect()) {
		cout << "Connected successfully!" << endl;
	} else {
		cout << "Could not connect!" << endl;
	}

	cout << (packageId = docDb.addPackage("name", "filename", "url")) << endl;
	cout << (sourceId = docDb.addSource(packageId, "method", "int", "test::test1", "return 0;")) << endl;
	cout << (namespaceId = docDb.addSource(packageId, "namespace", "", "scope", "")) << endl;
	cout << docDb.addDocumentation(packageId, sourceId, "Documentation1") << endl;
	cout << packageId << " =? " << docDb.getPackageIdFromName("name") << endl;
	cout << sourceId << " =? " << docDb.getSourceIdFromName(packageId, "test::test1") << endl;
	if (docDb.updateSource(packageId, sourceId, "return 1;"))
		cout << "Update succeeded!" << endl;
	else
		cout << "Update failed!\n" << endl;

	cout << docDb.addParameter(packageId, sourceId, "p1", "n1") << endl;

	cout << docDb.addDependencyName(packageId, sourceId, "scope") << endl;
	cout << docDb.addDependencyName(packageId, sourceId, "scape") << endl;
	
	cout << (namespaceId = docDb.addSource(packageId, "namespace", "", "scape", "")) << endl;
	cout << docDb.updateDependency("scape", namespaceId) << endl;
	docDb.disconnect();

	return 1;
}

File: languages/cpp/doxygen/docdb-test.pl
#!/usr/bin/perl

use lib "./";
use docdb;

$db = new docdb("localhost", "ir", "ir", "ir");

if (!$db->connect()) {
	die "Not connected!\n";
}
print "Connected!\n";

$new_package_id = $db->add_package("test-perl1", "test-perl2", "test-perl3");
$new_source_id = $db->add_source($new_package_id, "method", "return", "name", 500, "code");
$new_parameter_id = $db->add_parameter($new_package_id, $new_source_id, "type", "name");
#$db->update_source($new_package_id, $new_source_id, "code2");

print "new_package_id: $new_package_id \n";
print "new_source_id: $new_source_id \n";
print "new_parameter_id: $new_parameter_id \n";

$db->disconnect();
